src/types/tuple.rs

```diff
@@ -81,10 +84,44 @@ impl Tuple {
             } else {
                 let mut value_bytes = value.to_raw();
 
-                if value.is_variable() {
-                    bytes.append(&mut (value_bytes.len() as u32).encode_fixed_vec());
+                let col = &self.columns[i];
+                match col.desc.column_datatype {
+                    LogicalType::Boolean
+                    | LogicalType::Tinyint
+                    | LogicalType::UTinyint
+                    | LogicalType::Smallint
+                    | LogicalType::USmallint
+                    | LogicalType::Integer
+                    | LogicalType::UInteger
+                    | LogicalType::Bigint
+                    | LogicalType::UBigint
+                    | LogicalType::Decimal(_, _)
+                    | LogicalType::Float
+                    | LogicalType::Double
+                    | LogicalType::Date
+                    | LogicalType::DateTime => bytes.append(&mut value_bytes),
+                    LogicalType::Varchar(_) => {
+                        bytes.append(&mut (value_bytes.len() as u32).encode_fixed_vec());
+                        bytes.append(&mut value_bytes);
+                    }
+                    LogicalType::Text => {
+                        // len of text file
+                        let len = value_bytes.len();
+                        bytes.append(&mut (len as u32).encode_fixed_vec());
+                        // 6 bytes prefix of text content
+                        bytes.append(&mut value_bytes[..min(6, len)].to_vec());
+                        // text file absolute path
+                        let file_path = generate_text_file_path(
+                            self.id.as_ref().expect("The tuple has no primary key"),
```

Comment:
I don't think this is a recoverable error as binder should protect table must have primary key.

Reply:
> TypeError::PrimaryKeyNotFound

Using `TypeError::PrimaryKeyNotFound` can focus on dealing with the problem of no primary key, otherwise it will require such a long processing logic every time to obtain the id data, which is not elegant.